vault/router.go

```diff
@@ -236,8 +236,9 @@ func (r *Router) routeCommon(req *logical.Request, existenceCheck bool) (*logica
 	// Hash the request token unless this is the token backend
 	clientToken := req.ClientToken
 	switch {
-	case strings.HasPrefix(original, "auth/token/"):
-	case strings.HasPrefix(original, "cubbyhole/"):
+	case strings.HasPrefix(originalPath, "auth/token/"):
+	case strings.HasPrefix(originalPath, "sys/"):
```

Comment:
I am not able to recall why ClientToken was not passed to the logical backends, probably with an intention to disallow actions to be taken on behalf of the client. Clearly we are doing that for a few legitimate cases now. I wonder if it is a good idea to allow it on a case by case basis to select paths as opposed to making it available to all the endpoint of the system backend. Thoughts?